File: SportsPlatform.Auth/SportsPlatform.Auth.Api/wwwroot/react_web/src/services/invitationService.js
import api from './api';

// GET /invitations/:token — preview an invitation
export const getInvitation = async (token) => {
  const { data } = await api.get(`/invitations/${token}`);
  return data;
};

// POST /invitations/:token/accept — accept an invitation
export const acceptInvitation = async (token) => {
  const { data } = await api.post(`/invitations/${token}/accept`);
  return data;
};

// POST /clubs/:clubId/teams/:teamId/invitations — create team invitation
export const createTeamInvitation = async (clubId, teamId, inviteData) => {
  const { data } = await api.post(`/clubs/${clubId}/teams/${teamId}/invitations`, inviteData);
  return data;
};

// GET /clubs/:clubId/teams/:teamId/invitations — list team invitations
export const getTeamInvitations = async (clubId, teamId) => {
  const { data } = await api.get(`/clubs/${clubId}/teams/${teamId}/invitations`);
  return data;
};

// DELETE /clubs/:clubId/teams/:teamId/invitations/:invitationId — cancel team invitation
export const cancelTeamInvitation = async (clubId, teamId, invitationId) => {
  const { data } = await api.delete(`/clubs/${clubId}/teams/${teamId}/invitations/${invitationId}`);
  return data;
};

File: SportsPlatform.Auth/SportsPlatform.Auth.Api/wwwroot/eq-master/lib/services/invitation_service.dart
import '../models/api_models.dart';
import 'api_client.dart';

class InvitationService {
  final ApiClient _api = ApiClient.instance;

  Future<List<InvitationDto>> getMyInvitations() async {
    final json = await _api.get('/invitations/me');
    return (json as List)
        .map((e) => InvitationDto.fromJson(Map<String, dynamic>.from(e as Map)))
        .toList();
  }

  Future<dynamic> acceptInvitation(String token) {
    return _api.post('/invitations/${Uri.encodeComponent(token)}/accept');
  }

  Future<void> denyInvitation(String token) async {
    await _api.post('/invitations/${Uri.encodeComponent(token)}/deny');
  }

  Future<InvitationDto> createTeamInvitation(
      String clubId, String teamId, Map<String, dynamic> request) async {
    final json = await _api.post(
      '/clubs/$clubId/teams/$teamId/invitations',
      body: request,
    );
    return InvitationDto.fromJson(Map<String, dynamic>.from(json as Map));
  }

  Future<InvitationDto> createClubInvitation(
      String clubId, Map<String, dynamic> request) async {
    final json = await _api.post(
      '/clubs/$clubId/invitations',
      body: request,
    );
    return InvitationDto.fromJson(Map<String, dynamic>.from(json as Map));
  }

  Future<List<InvitationDto>> getTeamInvitations(
      String clubId, String teamId) async {
    final json = await _api.get('/clubs/$clubId/teams/$teamId/invitations');
    return (json as List)
        .map((e) => InvitationDto.fromJson(Map<String, dynamic>.from(e as Map)))
        .toList();
  }

  Future<void> deleteTeamInvitation(
      String clubId, String teamId, String invitationId) async {
    await _api.delete(
      '/clubs/$clubId/teams/$teamId/invitations/$invitationId',
    );
  }
}
